pennylane_cirq/simulator_device.py

```diff
@@ -198,95 +216,84 @@ def marginal_probability(self, wires):
 
         return marginal_probabilities
 
+    def _get_eigenvalues(self, observable, wires, par):
+        """Return the eigenvalues of the given observable.
 
-    def expval(self, observable, wires, par):
+        Args:
+            observable (str or list[str]): name of the observable(s)
+            wires (List[int] or List[List[int]]): subsystems the observable(s) is to be measured on
+            par (tuple or list[tuple]]): parameters for the observable(s)
+
+        Returns:
+            array[float]: eigenvalues of the observable
+        """
         num_wires = len(wires)
 
-        eigenvalues = np.ones(2**num_wires)
+        # All ones corresponds to the Identity observable
+        eigenvalues = np.ones(2 ** num_wires)
 
         if observable == "Hermitian":
             # Take the eigenvalues from the stored values
             Hmat = par[0]
             Hkey = tuple(Hmat.flatten().tolist())
-            eigenvalues = self._eigs[Hkey]["eigval"]
 
+            eigenvalues = self._eigs[Hkey]["eigval"]
         elif observable != "Identity":
-            # TODO: Add support for Tensor observables after it is merged in PL
-            eigenvalues[1] = -1
+            # If we don't have an Hermitian observable we use
+            # a diagonalization to tensors of Z observables
+            eigenvalues = z_eigs(num_wires)
+
+        return eigenvalues
+
+    def expval(self, observable, wires, par):
+        eigenvalues = self._get_eigenvalues(observable, wires, par)
 
         if self.analytic:
             # We have to use the state of the simulation to find the expectation value
-            marginal_probability = np.fromiter(self.marginal_probability(wires).values(), dtype=np.float)
+            marginal_probability = np.fromiter(
+                self.marginal_probability(wires).values(), dtype=np.float
+            )
 
             return np.dot(eigenvalues, marginal_probability)
         else:
             return self.sample(observable, wires, par).mean()
 
     def var(self, observable, wires, par):
-        wire = wires[0]
-
-        zero_value = 1
-        one_value = -1
-
-        if observable == "Hermitian":
-            # Take the eigenvalues from the stored values
-            Hmat = par[0]
-            Hkey = tuple(Hmat.flatten().tolist())
-            zero_value = self._eigs[Hkey]["eigval"][0]
-            one_value = self._eigs[Hkey]["eigval"][1]
-
-        elif observable == "Identity":
-            one_value = 1
+        eigenvalues = self._get_eigenvalues(observable, wires, par)
```

Comment:
This is great @johannesjmeyer! I think it's cleaner than the Qiskit structure. I'm thinking that since so many plugins are landing on using the same methods/functions, we should move them to PennyLane as a convenience.

We could even have a `QubitDevice` base class in PennyLane, that implements the 'agnostic' logic. Things like the `rotate_circuit`, which calls `dev.apply()` to rotate wires pre-measurement.

Reply:
Yes, I think too that there is some unified structure emerging, especially in the analytic plugins. The process of going from the state vector/probabilities to expectation values, variances and samples is essentially the same. 

But I would not (yet) create a new kind of `Device` for this. If we offer some convenience functions like `utils.expval(probabilities, eigenvalues)` then we can have a unified structure where it is appropriate. 

For the rotations, I would probably add them to the `Observables` themselves as `diagonalize()` (creating PL operations like the MottonenStatePrep) or something alike, so that plugins can call them if they only support Z expectations. But I would not create an automatism because that may not be appropriate for some plugins.